framework/src/modules/keys/keys_module.ts

```diff
@@ -38,4 +53,181 @@ export class KeysModule extends BaseModule {
 			numberOfSignatures: 0,
 		},
 	};
+
+	// eslint-disable-next-line class-methods-use-this
+	public async beforeTransactionApply({
+		stateStore,
+		transaction,
+	}: TransactionApplyInput): Promise<void> {
+		const sender = await stateStore.account.get<AccountKeyAsset>(transaction.senderID);
+		const { networkIdentifier } = stateStore.chain;
+		const transactionBytes = transaction.getSigningBytes();
+
+		const transactionWithNetworkIdentifierBytes = Buffer.concat([
+			networkIdentifier,
+			transactionBytes,
+		]);
+
+		// This is for registration of multisignature that requires all signatures
+		if (transaction.moduleType === this.type && transaction.assetType === RegisterAssetType) {
+			const { mandatoryKeys, optionalKeys } = codec.decode<DecodedAsset>(
+				KeysSchema,
+				transaction.asset,
+			);
+
+			// For multisig registration we need all signatures to be present
+			const numberOfExpectedKeys = mandatoryKeys.length + optionalKeys.length + 1;
+			if (numberOfExpectedKeys !== transaction.signatures.length) {
+				throw new Error(
+					`There are missing signatures. Expected: ${numberOfExpectedKeys} signatures but got: ${transaction.signatures.length}`,
+				);
+			}
+
+			// Check if empty signatures are present
+			if (!transaction.signatures.every(signature => signature.length > 0)) {
+				throw new Error('A signature is required for each registered key.');
+			}
+
+			// Verify first signature is from senderPublicKey
+			validateSignature(
+				transaction.senderPublicKey,
+				transaction.signatures[0],
+				transactionWithNetworkIdentifierBytes,
+				transaction.id,
+			);
+
+			// Verify each mandatory key signed in order
+			validateKeysSignatures(
+				mandatoryKeys,
+				transaction.signatures.slice(1, mandatoryKeys.length + 1),
+				transactionWithNetworkIdentifierBytes,
+				transaction.id,
+			);
+
+			// Verify each optional key signed in order
+			validateKeysSignatures(
+				optionalKeys,
+				transaction.signatures.slice(mandatoryKeys.length + 1),
+				transactionWithNetworkIdentifierBytes,
+				transaction.id,
+			);
+			return;
+		}
+
+		if (!isMultisignatureAccount(sender)) {
+			validateSignature(
```

Comment:
Maybe issed in the original base transaction, but shouldnt it check if `signatures.length` === 1?

Reply:
Added the check.